subsys/mgmt/mcumgr/lib/cmd/img_mgmt/src/zephyr_img_mgmt.c

```diff
@@ -236,28 +247,43 @@ img_mgmt_vercmp(const struct image_version *a, const struct image_version *b)
 }
 
 int
-img_mgmt_impl_erase_slot(void)
+img_mgmt_impl_erase_slot(int slot)
 {
+	const struct flash_area *fa;
+	int rc;
+	int area_id = zephyr_img_mgmt_flash_area_id(slot);
 	bool empty;
-	int rc, best_id;
 
-	/* Select any non-active, unused slot */
-	best_id = img_mgmt_get_unused_slot_area_id(-1);
-	if (best_id < 0) {
-		return MGMT_ERR_ENOENT;
+	if (area_id < 0) {
+		return MGMT_ERR_EUNKNOWN;
+	}
+
+	rc = flash_area_open(area_id, &fa);
+
+	if (rc != 0) {
+		return MGMT_ERR_EUNKNOWN;
 	}
-	rc = zephyr_img_mgmt_flash_check_empty(best_id, &empty);
+
+	rc = zephyr_img_mgmt_flash_check_empty_inner(fa, &empty);
+
 	if (rc != 0) {
 		return MGMT_ERR_EUNKNOWN;
 	}
 
 	if (!empty) {
-		rc = boot_erase_img_bank(best_id);
+	        rc = flash_area_erase(fa, 0, fa->fa_size);
+
 		if (rc != 0) {
 			return MGMT_ERR_EUNKNOWN;
 		}
 	}
 
+	flash_area_close(fa);
```

Comment:
This close will be missed if line 274 fails.

Reply:
Fixed